### PR TITLE
Updated SDK documentation and Updated Reservation API version to previous SDK release version.

### PR DESCRIPTION
**Pull request by rahuls-microsoft**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ x] Other, please clarify - Updated documentation. Updated Reservation API version to previous SDK release version.
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [x ] SDK of .NET (need service team to ensure code readiness)
      - [x ] SDK of Python
      - [ ] SDK of Java
      - [x ] SDK of Js
      - [x ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [ x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [ x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/reservations/resource-manager/Microsoft.Capacity/stable/2020-10-25/quota.json [hunk 1]
```diff
@@ -39,7 +39,7 @@
         "tags": [
           "QuotaInformation"
         ],
-        "description": "Get the current quotas (service limits) and usage of a resource. You can use the response from the GET operation to submit quota update requests.",
+        "description": "Get the current quota (service limits) and usage of a resource. You can use the response from the GET operation to submit quota update request.",
```
Comment (pilara-ms) [inline specification/reservations/resource-manager/Microsoft.Capacity/stable/2020-10-25/quota.json:42]: (service limits) should be (service limit)

File: specification/reservations/resource-manager/Microsoft.Capacity/stable/2020-10-25/quota.json [hunk 2]
```diff
@@ -285,7 +285,7 @@
             "$ref": "./examples/getQuotaRequestStatusFailed.json"
           }
         },
-        "description": "For the specified Azure region (location), get the details and status of the quota request by the quota request ID for the resources of the resource provider. The PUT request for service limits returns a response with the requestId parameter.",
+        "description": "For the specified Azure region (location), get the details and status of the quota request by the quota request ID for the resources of the resource provider. The PUT request for the quota (service limit) returns a response with the requestId parameter.",
```
Comment (pilara-ms) [inline specification/reservations/resource-manager/Microsoft.Capacity/stable/2020-10-25/quota.json:288]: Updated line 288 looks good.

Comment (rahuls-microsoft) [reply specification/reservations/resource-manager/Microsoft.Capacity/stable/2020-10-25/quota.json:288]: Great! thanks.

File: specification/reservations/resource-manager/Microsoft.Capacity/stable/2020-10-25/quota.json (hunk 1 above)
Comment (rahuls-microsoft) [reply specification/reservations/resource-manager/Microsoft.Capacity/stable/2020-10-25/quota.json]: Updated it.

File: specification/reservations/resource-manager/Microsoft.Capacity/stable/2020-10-25/quota.json [hunk 3]
```diff
@@ -39,7 +39,7 @@
         "tags": [
           "QuotaInformation"
         ],
-        "description": "Get the current quota (service limits) and usage of a resource. You can use the response from the GET operation to submit quota update request.",
+        "description": "Get the current quota (service limit) and usage of a resource. You can use the response from the GET operation to submit quota update request.",
```
Comment (pilara-ms) [inline specification/reservations/resource-manager/Microsoft.Capacity/stable/2020-10-25/quota.json:42]: Looks good. Thanks.

**Comment by rahuls-microsoft** [comment]
Why the Go SDK build is failing and blocking all other SDK generation?
Please suggest.

**Comment by weidongxu-microsoft** [comment]
@rahuls-microsoft I think it complains that preview json `preview/2020-10-01-preview/reservations.json` should be in tag labeled with `-preview`.

@ArcturusZhang for insight on Go error. We seems not requiring above for all SDKs?

File: specification/reservations/resource-manager/readme.md [hunk 4]
```diff
@@ -35,7 +35,7 @@ These settings apply only when `--tag=package-2020-10-25` is specified on the co
 
 ``` yaml $(tag) == 'package-2020-10-25'
 input-file:
-  - Microsoft.Capacity/stable/2017-11-01/reservations.json
+  - Microsoft.Capacity/preview/2020-10-01-preview/reservations.json
```
Comment (ArcturusZhang) [inline specification/reservations/resource-manager/readme.md:38]: Changing this makes this tag `package-2020-10-25` a preview package, which results a error in go SDK package validation. In go SDK, a preview package has the different patterns of path from the stable packages.
Is this change really necessary? If not, could we revert this change? 
If this is necessary, I recommend that we make a new tag for the preview swagger.

**Review by ArcturusZhang — CHANGES_REQUESTED** [review]

File: specification/reservations/resource-manager/readme.md (hunk 4 above)
Comment (rahuls-microsoft) [reply specification/reservations/resource-manager/readme.md:38]: Yes, we need to include the right version of the Reservation SDK, as the previous release of SDK was - preview/2020-10-01-preview, so we cannot go back and it breaks all the tests built with the SDK.

If I create a tag ``` yaml $(tag) == 'package-2020-10-25-preview' for Go, as it's giving error and use package-2020-10-25 for others, will it work?

Comment (rahuls-microsoft) [reply specification/reservations/resource-manager/readme.md:38]: Added following config to Readme.go.md 
### Tag: package-2020-10-25 and go

These settings apply only when `--tag=package-2020-10-25 --go` is specified on the command line.
Please also specify `--go-sdk-folder=<path to the root directory of your azure-sdk-for-go clone>`.

``` yaml $(tag)=='package-2020-10-25' && $(go)
output-folder: $(go-sdk-folder)/services/preview/$(namespace)/mgmt/2020-10-25/$(namespace)
```
 I have added /preview path for package-2020-10-25 in readme.go.md .

Comment (rahuls-microsoft) [reply specification/reservations/resource-manager/readme.md:38]: Thanks for the help. Go SDk is building fine. Please merge the changes.

**Review by weidongxu-microsoft — APPROVED** [review]

**Comment by weidongxu-microsoft** [comment]
@ArcturusZhang RP updated, approve if you are OK with it.

**Comment by rahuls-microsoft** [comment]
Please approve and merge.
Thanks.

File: specification/reservations/resource-manager/readme.md (hunk 4 above)
Comment (ArcturusZhang) [reply specification/reservations/resource-manager/readme.md:38]: Well... this change resolves the CI failure in this PR, but it introduces a big breaking change to go SDK. 
By changing the package path in readme.go.md, we are renaming one package from stable to preview, a.k.a. removing one stable package and adding a new preview package. This would be considered as a breaking change.
Some customers might have concerns of using preview packages because breaking changes might be introduced to it in any time.
Is it possible that we keep this tag `package-2020-10-25` unchanged, and introduce a new tag with this new preview swagger? This would leave the old package unchanged.

**Review by ArcturusZhang — CHANGES_REQUESTED** [review]

File: specification/reservations/resource-manager/readme.md (hunk 4 above)
Comment (rahuls-microsoft) [reply specification/reservations/resource-manager/readme.md:38]: The previously released package was using the same preview version.
The last stable version was released in 2017. Customers have already moved to preview versions, so there shouldn't be any impact. 
The customers have been using the previous preview version of package, as the last stable version release was in 2017. Going back to 2017 version of reservation API,  breaks all the packages and the tests associated with it. And it doesn't help customers, as it will break their automation, as they have moved forward. That would be breaking change.

There is not much change from previous version of package and current package-
1. Version no updated and preview feature API is removed.

**The tag package-2020-10-25 is introduced for this GA release. It doesn't exists or released in any form. It's new package getting created.**

Please proceed forward with the merge, so that we can go GA by 1/19.
Thanks for help for time release of SDKs.

**Review by ArcturusZhang — APPROVED** [review]
